File: 0014-longest-common-prefix/0014-longest-common-prefix.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        StringBuilder sb = new StringBuilder();
        int minLength = Integer.MAX_VALUE;
        String minStr = null;
        
        for (String str : strs) {
            if (str.length() < minLength) {
                minLength = str.length();
                minStr = str;
            } 
        }
        
        int idx = 0;
        
        while (idx < minLength) {
            boolean same = true;
            Character c = null;
            for (String str : strs) {
                c = minStr.charAt(idx);
                if (!c.equals(str.charAt(idx))) {
                    same = false;
                    break;
                }
            }    
            if (!same) {
                break;
            }
            sb.append(c);
            idx += 1;
        }
        
        return sb.toString();
    }
}

File: 0014-longest-common-prefix/0014-longest-common-prefix.kt
class Solution {
    fun longestCommonPrefix(strs: Array<String>): String {
        var sb: StringBuilder  = StringBuilder()
        var minLength: Int = Int.MAX_VALUE
        var minStr: String = ""
        
        for (str in strs) {
            if (str.length < minLength) {
                minLength = str.length
                minStr = str
            } 
        }
        
        var idx: Int = 0
        
        while (idx < minLength) {
            var same = true
            var c = minStr[idx]
            for (str in strs) {
                if (!c.equals(str[idx])) {
                    same = false
                    break
                }
            }    
            if (!same) {
                break
            }
            sb.append(c)
            idx += 1
        }
        
        return sb.toString()
    }
}